src/editor/EditorWindow.cpp

```diff
@@ -229,6 +230,17 @@ EditorWindow::EditorWindow(QWidget *parent)
   connect(ui->actionCopyOptions, &QAction::triggered, [this]() {
     m_copy_options_dialog->setVisible(!m_copy_options_dialog->isVisible());
   });
+
+  if(Settings::ShowLandingPage::get())
+    {
+      m_welcome_dialog = new WelcomeDialog;
```

Comment:
Most of the other dialogs are created in the constructor of EditorWindow, is there a particular reason this one is created here?

Reply:
that is the editorwindow constructor

Reply:
oops, my bad, I didn't see that lol. I think it might be a bit more defensive if the welcome dialog were instantiated in the initializer list so that even if the setting isn't set it's ready for use (e.g., if we later add a way to show the welcome window via a menu item or something)

Reply:
i've moved it to where most of the other things are defined, not sure if that's where you meant by initializer list (beginning of the constructor) 
showing the welcome dialog afterwards is easy, ive already made a slot that shows it for the QTimer to use (`EditorWindow::showWelcomeDialog()`) if you actually did want to do that

Reply:
sorry I think I was unclear, what I meant is that if Settings::ShowLandingPage::get() is false on startup then we have to be careful in the future if we want to call showWelcomeDialog in more contexts because m_welcome_dialog would not be initialized. so, I think it would be a bit more future-proof if something like `m_welcome_dialog(new WelcomeDialog(this))` was added to the initializer list (see where m_host_dialog is initialized). the other lines can go anywhere as far as i'm concerned. does that make sense?

Reply:
oh so what you're saying is i should just initialize the dialog and only have the setting control whether it's shown or not? that sounds fine to me, but there shouldnt be any need as any way that instance of WelcomeDialog is currently accessed is blocked by a check for the setting
that's done